crates/next-core/js/src/entry/router.ts

```diff
@@ -6,6 +6,7 @@ import loadNextConfig from "@vercel/turbopack-next/entry/config/next";
 import { makeResolver } from "next/dist/server/router.js";
 
 import "next/dist/server/node-polyfill-fetch.js";
+import * as middleware from "MIDDLEWARE_CONFIG";
```

Comment:
Can you explain why this is beneficial? How is middleware executed in a different way?

Reply:
It's executed "on the edge". Next.js has some emulation for that via `vm`. There are different globals available in middleware and you can't use `require` etc.

Reply:
I'm still not understanding why this is necessary. We're not using `require` here, this will be transformed into the same `__turbopack_import__('[project-with-next]/my-app/middleware.ts (ecmascript)')` as every other import in this file. I don't understand why importing the middleware file requires special behavior.